### PR TITLE
db, server: Aggregate illegal reports into /all/

### LINKED ISSUE #571: Aggregate all illegal reports into one page
We can use `/html/reports/all` for that.

### PR DESCRIPTION
**Pull request by Chiiruno**
Fixes #571

### CODE REVIEW

File: db/report.go [hunk 1]
```diff
@@ -25,25 +26,41 @@ func Report(id uint64, board, reason, ip string, illegal bool) error {
...
+
 	err = queryAll(
 		sq.Select("id", "target", "reason", "created").
 			From("reports").
-			Where("board = ?", board).
+			Where(where).
 			OrderBy("created desc"),
 		func(r *sql.Rows) (err error) {
 			err = r.Scan(&tmp.ID, &tmp.Target, &tmp.Reason, &tmp.Created)
+
```
Comment (bakape) [inline db/report.go]: Nope

File: db/report.go [hunk 2]
```diff
@@ -25,25 +26,41 @@ func Report(id uint64, board, reason, ip string, illegal bool) error {
...
 			if err != nil {
 				return
 			}
+
```
Comment (bakape) [inline db/report.go]: Nope

File: db/report.go [hunk 3]
```diff
@@ -25,25 +26,41 @@ func Report(id uint64, board, reason, ip string, illegal bool) error {
...
 			rep = append(rep, tmp)
 			return
 		},
 	)
+
```
Comment (bakape) [inline db/report.go]: Nope

File: server/report.go [hunk 4]
```diff
@@ -88,17 +88,26 @@ func reportForm(w http.ResponseWriter, r *http.Request) {
 
 // Render a list of reports for the board
 func reportList(w http.ResponseWriter, r *http.Request) {
+	var rep []auth.Report
+	var err error
 	board := extractParam(r, "board")
-	if !auth.IsNonMetaBoard(board) {
+
```
Comment (bakape) [inline server/report.go]: Nope

File: server/report.go [hunk 5]
```diff
@@ -88,17 +88,26 @@ func reportForm(w http.ResponseWriter, r *http.Request) {
...
+		rep, err = db.GetReports(board, true)
+	} else {
+		rep, err = db.GetReports(board, false)
+	}
+
 	if err != nil {
 		httpError(w, r, err)
 		return
 	}
+
```
Comment (bakape) [inline server/report.go]: Nope

File: server/report.go [hunk 6]
```diff
@@ -88,17 +88,26 @@ func reportForm(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
-	rep, err := db.GetReports(board)
+	if board == "all" {
+		rep, err = db.GetReports(board, true)
+	} else {
+		rep, err = db.GetReports(board, false)
+	}
+
```
Comment (bakape) [inline server/report.go]: `rep, err := db.GetReports(board, board == 'all')`

File: db/report.go [hunk 7]
```diff
@@ -25,25 +26,41 @@ func Report(id uint64, board, reason, ip string, illegal bool) error {
...
 	tmp := auth.Report{
 		Board: board,
 	}
-	rep = make([]auth.Report, 0, 64)
+
+	if illegal {
+		if board == "all" {
+			where = "illegal = true"
+		} else {
+			where = fmt.Sprintf("board = '%s' and illegal = true", board)
```
Comment (bakape) [inline db/report.go]: Is this branch ever actually called?

File: server/report.go (hunk 6 above)
Comment (Chiiruno) [reply server/report.go]: That's the power of sleep deprivation for you!

File: db/report.go (hunk 7 above)
Comment (Chiiruno) [reply db/report.go]: Not currently, however it could be used for a staff reports live popup, or notification.

Comment (bakape) [reply db/report.go]: If it is not used currently, remove it. Don't plan for contingencies too much. It leads to messy code.

**Comment by bakape** [comment]
Please don't rebase the branch. I can't view the changes and have to review the entire thing all over then.

**Pull request closed without merging**